src/components/Application/Navigation.tsx

```diff
@@ -0,0 +1,133 @@
+import { Button, Segment, Sticky } from 'semantic-ui-react'
+import {
+  MethodRevalidate,
+  MethodToCallProps,
+  SectionAndPage,
+  SectionDetails,
+  SectionsStructureNEW,
+} from '../../utils/types'
+import strings from '../../utils/constants'
+import { useRouter } from '../../utils/hooks/useRouter'
+
+interface NavigationProps {
+  current: SectionAndPage
+  sections: SectionsStructureNEW
+  serialNumber: string
+  requestRevalidation: MethodRevalidate
+}
+
+const Navigation: React.FC<NavigationProps> = ({
+  current,
+  sections,
+  serialNumber,
+  requestRevalidation,
+}) => {
+  const { push } = useRouter()
+
+  const currentSectionDetails = sections[current.sectionCode].details
+
+  const nextSection = (): SectionDetails | null => {
+    const nextSections = Object.values(sections)
+      .filter(({ details: { index } }) => index > currentSectionDetails.index)
+      .sort(({ details: { index: aIndex } }, { details: { index: bIndex } }) => aIndex - bIndex)
+    return nextSections.length > 0 ? nextSections[0].details : null
+  }
+
+  const previousSection = (): SectionDetails | null => {
+    const previousSections = Object.values(sections)
+      .filter(({ details: { index } }) => index < currentSectionDetails.index)
+      .sort(({ details: { index: aIndex } }, { details: { index: bIndex } }) => bIndex - aIndex)
+    return previousSections.length > 0 ? previousSections[0].details : null
+  }
+
+  const isFirstPage = current.pageNumber - 1 === 0 && previousSection() == null
+  const isLastPage =
+    current.pageNumber + 1 > currentSectionDetails.totalPages && nextSection() == null
+
+  const getPreviousSectionPage = (): SectionAndPage => {
+    const { sectionCode, pageNumber } = current
+    if (pageNumber > 1) return { sectionCode, pageNumber: pageNumber - 1 }
+    return {
+      sectionCode: (previousSection() as SectionDetails).code,
+      pageNumber: (previousSection() as SectionDetails).totalPages,
+    }
+  }
+
+  const getNextSectionPage = (): SectionAndPage => {
+    const { sectionCode, pageNumber } = current
+    if (pageNumber < currentSectionDetails.totalPages)
+      return { sectionCode, pageNumber: pageNumber + 1 }
+    return {
+      sectionCode: (nextSection() as SectionDetails).code,
+      pageNumber: 1,
+    }
+  }
+
+  const sendToPage = (sectionPage: SectionAndPage) => {
+    const { sectionCode, pageNumber } = sectionPage
+    push(`/applicationNEW/${serialNumber}/${sectionCode}/Page${pageNumber}`)
+  }
+
+  const previousButtonHandler = (_: any) => {
+    const previousSectionPage = getPreviousSectionPage()
+    sendToPage(previousSectionPage)
+  }
+
+  const nextPageButtonHandler = (_: any) => {
+    const nextSectionPage = getNextSectionPage()
+    // Use validationMethod to check if can change to page (on linear application) OR
+    // display current page with strict validation
+    requestRevalidation(({ firstStrictInvalidPage, setStrictSectionPage }: MethodToCallProps) => {
```

Comment:
Need to check for `isLinear`, and only requestRevalidation if true.

Reply:
Done: 3aac044